frame/support/procedural/src/lib.rs

```diff
@@ -778,6 +781,243 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
 		.into()
 }
 
+/// This attribute can be used to derive a full implementation of a trait based on a local
+/// partial impl and an external impl containing defaults that can be overriden in the local
+/// impl.
+///
+/// ## Usage
+///
+/// The attribute should be attached to an impl block (strictly speaking a `syn::ItemImpl`) for
+/// which we want to inject defaults in the event of missing trait items in the block.
+///
+/// The attribute takes two arguments separated by an `as` with the general form:
```

Comment:
We don't support changing the name of `DefaultConfig` currently? So we can assume this is always the name right now? 

I would really like to get rid off this `as`  and only have it optionally.

Reply:
Well the reason being, these docs describe `derive_impl`, which can be used for other things, not just this default config stuff. Originally I wrote docs for this that were 100% specific to this use case, but @kianenigma  pointed out that this is more general and so we made 2-3 changes to make sure this could be used for arbitrary things as demonstrated in the UI tests

Reply:
I think you misread what Basti said, he meant that the docs right now reads as if `as` is a must, when it's actually supposed to be optional.

Reply:
ahhh, kk